docs/release/release-tasks.md

```diff
@@ -0,0 +1,384 @@
+* Release lead should keep an eye on what is going on in the project to be able to react if necessary
+
+### Tasks
+
+#### Set a tentative release date for the minor release
+
+1. Set a tentative release date for the release and document it by creating a `release-1.4.md`.
+
+#### Assemble release team
+
+There is currently no formalized process to assemble the release team.
+As of now we ask for volunteers in Slack and office hours.
+
+#### Finalize release schedule and team
+
+1. Finalize release schedule and team in `release-1.4.md`.
+2. Update Slack user group and GitHub team accordingly.
+   <br>Prior art: https://github.com/kubernetes-sigs/cluster-api/issues/7476
+
+#### Prepare main branch for development of the new release
+
+The goal of this issue is to bump the versions on the main branch, so they are using 
+the upcoming release version and that our tests on main are testing the previous release.
+
+This comes down to changing occurrences of the old version to the new version (e.g. `v1.3` to `v1.4`):
+1. Add new release to top-level `metadata.yaml` and `test/e2e/data/shared/v1beta1/metadata.yaml`.
+2. Update providers in docker.yaml. Roughly:
+   1. Create a new entry for each provider for `v1.3.0`.
+   2. Rename `v1.3.99` to `v1.4.99`.
+3. Modify the test specs in `test/e2e/clusterctl_upgrade_test.go`.
+4. Update `create-local-repository.py` and `tools/tilt-prepare/main.go`: `v1.3.99` => `v1.4.99`.
+5. Make sure all tests are green (also run `pull-cluster-api-e2e-full-main` and `pull-cluster-api-e2e-workload-upgrade-1-23-latest-main`).
+
+Prior art: https://github.com/kubernetes-sigs/cluster-api/pull/6834/files
+
+#### Create a new GitHub milestone
+
+1. Create the milestone for the new release via GitHub UI.
+2. Update the [milestone_applier config](https://github.com/kubernetes/test-infra/blob/0b17ef5ffd6c7aa7d8ca1372d837acfb85f7bec6/config/prow/plugins.yaml#L371) accordingly.
+   <br>Prior art: [cluster-api: update milestone applier config for v1.3](https://github.com/kubernetes/test-infra/pull/26631)
+
+#### [Track] Remove previously deprecated code
+
+The goal of this task is to remove all previously deprecated code that can be now removed.
+
+1. Check for deprecated code and remove it.
+    * We can't just remove all code flagged with `Deprecated`. In some cases like e.g. in API packages
+      we have to keep the old code.
+
+Prior art: [Remove code deprecated in v1.2](https://github.com/kubernetes-sigs/cluster-api/pull/6779)
+
+#### [Track] Bump dependencies
+
+The goal of this task is to ensure that we have relatively up-to-date dependencies at the time of the release.
+This reduces the risk that CVEs are found in outdated dependencies after our release.
+
+We should take a look at the following dependencies:
+* Go dependencies in `go.mod` files.
+* Tools used in our Makefile (e.g. kustomize).
+
+#### Create a release branch
+
+1. Create the release branch locally based on the latest commit on main and push it:
```

Comment:
This will need to be updated after #7128 cc @ykakarap

Reply:
I'll add a general note to #7128 as I assume this PR will be merged before the automation